### PR TITLE
[Microsoft Generator CSharp] Generate properties for Model types

### PR DESCRIPTION
**Pull request by ShivangiReja**
Transferring this PR from [jorgerangel-msft/Microsoft.Generator.CSharp](https://github.com/jorgerangel-msft/Microsoft.Generator.CSharp/pull/51) to this repo.

This PR implements the `BuildProperties()` method within the `ModelTypeProvider` class and generate properties for model types.

- Fixes: https://github.com/Azure/autorest.csharp/issues/4475
- Fixes: https://github.com/Azure/autorest.csharp/issues/4520

### CODE REVIEW

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/mocks/tspCodeModel.json
Comment (timotheeguerin) [inline packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/mocks/tspCodeModel.json:1]: We are trying to not have a million line of snapshot code, can this be removed until a better solution is found. It is not going to be manageable to have every emitter with 100s of thousands of snapshot files that keep getting updated and crash GitHub ui

Comment (m-nash) [reply packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/mocks/tspCodeModel.json:1]: We only have one project now which is tiny compared to what we will see once all the cadl-ranch projects come in.

Comment (ShivangiReja) [reply packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/mocks/tspCodeModel.json:1]: We're adding this only once. We are using this for tests, so we cannot remove this.

**Review by m-nash — APPROVED** [review]

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/mocks/tspCodeModel.json
Comment (timotheeguerin) [reply packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/mocks/tspCodeModel.json:1]: that is my point its only gonna get worse

**Pull request merged**